src/index.ts

```diff
@@ -38,79 +27,114 @@ async function loadSubgraphSchemas(): Promise<ServiceDefinition[]> {
-        let artifactPath = '';
-        if (artifact.groupId !== undefined) {
-            artifactPath = `v1/groups/${artifact.groupId}/artifacts/${artifact.id}`
-        } else {
-            artifactPath = `v1/artifacts/${artifact.id}`
-        }
+    const res: Response<string> = await got(artifactPath, {
+        prefixUrl: prefix
+    });
 
-        const gqlRes: Response<string> = await got(artifactPath, {
-            prefixUrl: prefix
-        });
+     // TODO: DEBUG log here
+    return res;
+}
 
-        let url = '';
+function extractSubgraphUrl(artifact: Artifact){
 
-        if (artifact.labels !== undefined) {
-            for (const label of artifact.labels) {
-                if (label.startsWith('xjoin-subgraph-url=')) {
-                    url = label.split('xjoin-subgraph-url=')[1]
-                }
+    // TODO: DEBUG log here
+    let url = ''
+
+    if (artifact.labels !== undefined) {
+        for (const label of artifact.labels) {
+            if (label.startsWith('xjoin-subgraph-url=')) {
+                // TODO: DEBUG log here
+                url = label.split('xjoin-subgraph-url=')[1]
             }
         }
+    }
+
+    // TODO: DEBUG log here
+    return url
+}
+
+async function loadSubgraphSchemas(): Promise<ServiceDefinition[]> {
+     // TODO: INFO log here
+    const artifactsRespose : Response<ArtifactsResponse> = await fetchArtifacts();
+
+    const serviceDefinitions: ServiceDefinition[] = [];
+    const artifacts: Artifact[] = artifactsRespose.body.artifacts;
+
+    if (artifacts.length == 0) {
+        // TODO: INFO log here
+        return [];
+    }
+
+    for (const artifact of artifacts) {
+        // TODO: INFO log here
+        const gqlResponse: Response<string> = await fetchArtifactsDetails(artifact);
+
+        let url = extractSubgraphUrl(artifact);
 
         serviceDefinitions.push({
             name: artifact.id,
             url: url,
-            typeDefs: parse(gqlRes.body)
+            typeDefs: parse(gqlResponse.body)
         });
     }
 
     return serviceDefinitions;
 }
 
-const server = new ApolloServer({
-    introspection: true,
-    gateway: new ApolloGateway({
-        async supergraphSdl({update, healthCheck}) {
-            const poll = function () {
-                setTimeout(async () => {
-                    try {
-                        const compositionResult = await composeServices(await loadSubgraphSchemas())
-                        if (!compositionResult.errors) {
-                            // await healthCheck(compositionResult.supergraphSdl) //TODO parameterize so this is disabled in dev, enabled in prod
-                            update(compositionResult.supergraphSdl)
-                        }
-                    } catch (e) {
-                        console.error(e);
+const gateway = new ApolloGateway({
+    async supergraphSdl({update, healthCheck}) {
+        const poll = function () {
+
+            // IDEA: 
```

Comment:
I'd suggest to write ideas outside of codebase.

Reply:
removed from code